scripts/tools/IndexResourcePopulator.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoAdvancedSearch\scripts\tools;
+
+use oat\oatbox\extension\script\ScriptAction;
+use oat\oatbox\reporting\Report;
+use oat\tao\scripts\tools\MigrationAction;
+use oat\taoAdvancedSearch\model\Resource\Repository\IndexableClassRepository;
+use oat\taoAdvancedSearch\model\Resource\Repository\IndexableClassRepositoryInterface;
+use oat\taoAdvancedSearch\model\Resource\Task\ResourceMigrationTask;
+use Zend\ServiceManager\ServiceLocatorAwareInterface;
+use Zend\ServiceManager\ServiceLocatorAwareTrait;
+
+class IndexResourcePopulator extends ScriptAction implements ServiceLocatorAwareInterface
```

Comment:
Are you sure it's a good idea to declare return type for `provideUsage()`, `provideOptions()`, `provideDescription()` and `run()` methods in case these return types were not declared in `ScriptAction`?

Reply:
We have in the previous deleted IndexPopulator.php this and it was not an issue: 

https://github.com/oat-sa/extension-tao-advanced-search/pull/30/files#diff-a5038c7f6707e46b2df40107e09593b0968c45491998f084e9502d0e1c4d5803L55